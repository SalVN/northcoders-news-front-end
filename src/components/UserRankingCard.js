import React from 'react';
import PropTypes from 'prop-types';
import { BrowserRouter as Router, Link } from 'react-router-dom';

import './css/UserListCard.css';

const UserRankingCard = function (props) {
    let data;
    if (props.criteria === 'Sort by ranking') data = props.user.articles_vote_count + props.user.comments_vote_count;
    if (props.criteria === 'Sort by comments') data = props.user.comment_count;
    if (props.criteria === 'Sort by article votes') data = props.user.articles_vote_count;
    if (props.criteria === 'Sort by comment votes') data = props.user.comments_vote_count;
    return (
        <div className='container-fluid mini-card' key={props.user.username}>
            <div className='columns'>
                <div className='column is-1'>
                    <p className='user-list-card-number'><strong>{props.index + 1}</strong></p>
                </div>
                <div className='column is-two-thirds'>
                    <Link to={`/users/${props.user.username}`}>
                        <div>
                            <img className='user-list-image' src={`${props.user.avatar_url}`} />
                            <span className='user-list-username-text'>
                                {props.user.username}
                            </span>
                        </div>
                    </Link>
                </div>
                <div className='column is-3'>
                    <p className='user-list-card-ranking-number'>{data}</p>
                </div>
            </div>
        </div>
    );
};

UserRankingCard.propTypes = {
    user: PropTypes.object.isRequired,
    index: PropTypes.number.isRequired,
    criteria: PropTypes.string.isRequired
};

export default UserRankingCard;