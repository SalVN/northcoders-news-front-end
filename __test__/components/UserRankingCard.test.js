import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import UserRankingCard from '../../src/components/UserRankingCard';

describe('UserRankingCard', () => {
    const user = {
        _id: '59b1b18b327cce1fb043bdb1',
        username: 'northcoder',
        name: 'Awesome Northcoder',
        comment_count: 5,
        articles_vote_count: 20,
        comments_vote_count: 30,
        avatar_url: 'https://avatars3.githubusercontent.com/u/6791502?v=3&s=200',
        __v: 0
    };
    it('is a function', () => {
        expect(typeof UserRankingCard).toEqual('function');
    });

    it('renders', () => {
        const enzymeWrapper = shallow(<UserRankingCard
            user={user}
            index={0}
            criteria={'Sort by ranking'}
        />);
        expect(enzymeWrapper.children().length).toEqual(1);
    });

    it('renders correctly', () => {
        const tree = renderer.create(
            <MemoryRouter>
                <UserRankingCard
                    user={user}
                    index={0}
                    criteria={'Sort by ranking'}
                />
            </MemoryRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('contains one link', () => {
        const enzymeWrapper = shallow(<UserRankingCard
            user={user}
            index={0}
            criteria={'Sort by ranking'}
        />);
        expect(enzymeWrapper.find('Link').length).toBe(1);
    });

    it('renders the total of article votes and comment votes for the user if criteria === Sort by ranking', () => {
        const enzymeWrapper = shallow(<UserRankingCard
            user={user}
            index={0}
            criteria={'Sort by ranking'}
        />);
        expect(enzymeWrapper.find('.user-list-card-ranking-number').node.props.children).toBe(50);
    });

    it('renders the total number of comments  for the user if criteria === Sort by comments', () => {
        const enzymeWrapper = shallow(<UserRankingCard
            user={user}
            index={0}
            criteria={'Sort by comments'}
        />);
        expect(enzymeWrapper.find('.user-list-card-ranking-number').node.props.children).toBe(5);
    });

    it('renders the total article votes for the user if criteria === Sort by article votes', () => {
        const enzymeWrapper = shallow(<UserRankingCard
            user={user}
            index={0}
            criteria={'Sort by article votes'}
        />);
        expect(enzymeWrapper.find('.user-list-card-ranking-number').node.props.children).toBe(20);
    });

    it('renders the total comment votes for the user if criteria === Sort by comment votes', () => {
        const enzymeWrapper = shallow(<UserRankingCard
            user={user}
            index={0}
            criteria={'Sort by comment votes'}
        />);
        expect(enzymeWrapper.find('.user-list-card-ranking-number').node.props.children).toBe(30);
    });
});