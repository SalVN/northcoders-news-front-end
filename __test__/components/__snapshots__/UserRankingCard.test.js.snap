// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRankingCard renders correctly 1`] = `
<div
  className="container-fluid mini-card"
>
  <div
    className="columns is-mobile"
  >
    <div
      className="column is-1 is-hidden-tablet-only"
    >
      <p
        className="user-list-card-number"
      >
        <strong>
          1
        </strong>
      </p>
    </div>
    <div
      className="column is-two-thirds"
    >
      <a
        href="/users/northcoder"
        onClick={[Function]}
      >
        <div>
          <img
            className="user-list-image"
            src="https://avatars3.githubusercontent.com/u/6791502?v=3&s=200"
          />
          <span
            className="user-list-username-text"
          >
            northcoder
          </span>
        </div>
      </a>
    </div>
    <div
      className="column is-3"
    >
      <p
        className="user-list-card-ranking-number"
      >
        50
      </p>
    </div>
  </div>
</div>
`;
